common/build-style/gem.sh

```diff
@@ -0,0 +1,60 @@
+#
+# This helper is for templates using gem files from RubyGems.
+#
+do_install() {
+	: ${gem_cmd:=gem}
+
+	local _GEMDIR _INSTDIR
+	
+	_GEMDIR=$($gem_cmd env gemdir)
+	_INSTDIR=${DESTDIR}/${_GEMDIR}/gems/${pkgname#ruby-}-${version}
+
+	$gem_cmd install \
+		--local \
+		--install-dir ${DESTDIR}/${_GEMDIR} \
+		--bindir ${DESTDIR}/usr/bin \
+		--ignore-dependencies \
+		--no-document \
+		--verbose \
+		${XBPS_SRCDISTDIR}/${pkgname}-${version}/${pkgname#ruby-}-${version}.gem
+
+	# Remove cache
+	rm -rf ${DESTDIR}/${_GEMDIR}/cache
+
+	# Remove ext directory, they are only source code and configuration
+	# The actual extensions are in a arch path guarded
+	rm -rf ${_INSTDIR}/ext
+
+	# Remove installed tests and benchmarks
+	rm -rf ${_INSTDIR}/{test,tests,autotest,benchmark,benchmarks,script,examples,demo}
+
+	# Remove files shipped on the root of the gem, most of the time they are useless
```

Comment:
_most_ of the time?

Reply:
I have never met one that required it, root is mostly littered with the following:

- git files (.gitignore, .gitmodules)
- README (lol nobody reads that)
- LICENSE/COPYING (We put them in the proper place with vlicense if necessary)
- .gemspec file (Ruby requires them somewhere else)

Reply:
Ah, _most_ of the time just sounds as if they're important for some package but we just go ahead and delete them anyway.